docs/editors/new-editor.md

```diff
@@ -269,5 +274,95 @@ focus back to `UserTest.scala` that depends on APIs defined in `User.scala`.
 
 _Notification_:
 
-- method: `metals/didFocus`
+- method: `metals/didFocusTextDocument`
 - params: `string`, the URI of the document where the focused was moved to.
+
+## Metals server properties
+
+The Metals language server is configured through JVM system properties.
+Configuration options are added to allow customization for particular clients
+since editors interpret LSP functionality slightly differently. For example, in
+Vim the `window/logMessage` notification is always visible in the UI while in VS
+Code logs are hidden by default.
+
+### `-Dmetals.file-watcher`
+
+Possible values:
+
+- `auto` (default): use editor file watcher if supported, otherwise use Metals
+  built-in file watcher.
+- `custom`: the client will send `workspace/didChangeWatchedFiles` notifications
+  even if the editor doesn't declare in the `initialize` hand-shake that it
+  supports file watching.
```

Comment:
Even if the editor declares that it supports file watching on initialize, it still needs to accept the patterns through an additional dynamic capabilities registration request.

Reply:
Yes, I think this is what we should do. Because I don't see any other point in this initialization setting: if the client says it supports it, it's going to send the notification.